lib/JITIDL/Chakra.JITIDL.vcxproj

```diff
@@ -31,7 +31,9 @@
   <ItemGroup>
     <Midl Include="$(MsBuildThisFileDirectory)ChakraJIT.idl">
       <OutputDirectory>$(IntDir)</OutputDirectory>
-      <AdditionalOptions>%(AdditionalOptions) /acf "$(MsBuildThisFileDirectory)ChakraJIT.acf" /prefix client "Client" server "Server"</AdditionalOptions>
+      <!-- This is a messy work-around for a bug in clang and a bug in midl. Clang-cl purports to support a /P option to preprocess to a file, which does not work, so we have to find a different cl on the system. Midl makes us use short paths because it doesn't properly support spaces in arguments. -->
+      <AdditionalOptions Condition="'$(Clang)'!='ClangCL'">%(AdditionalOptions) /acf "$(MsBuildThisFileDirectory)ChakraJIT.acf" /prefix client "Client" server "Server"</AdditionalOptions>
+      <AdditionalOptions Condition="'$(Clang)'=='ClangCL'">%(AdditionalOptions) /acf "$(MsBuildThisFileDirectory)ChakraJIT.acf" /prefix client "Client" server "Server" /cpp_cmd "C:\Progra~2\MIB055~1\Preview\Enterprise\VC\Tools\MSVC\14.14.26310\bin\Hostx64\x64\cl.exe"</AdditionalOptions>
```

Comment:
For my own understanding, where did this path come from?

Reply:
This is very sloppy and the worst part of the PR. Currently this needs to be re-written on a per-system basis. Before any real attempt to merge this is made, I'm going to re-write this part, with the worst case being a script that attempts to figure out where the actual cl is. There's just too many bad constraints on this.

In terms of constructing the path, you can get windows short directory names with `dir /x`, which helps you work with tools that don't support spaces in arguments. It's a legacy artefact, but helps in situations like this one.

Reply:
Found a much better work-around for the new version, thankfully.